core/src/main/java/org/apache/spark/shuffle/api/ShuffleDriverComponents.java

```diff
@@ -66,8 +66,19 @@ default void removeShuffle(int shuffleId, boolean blocking) {}
    * Does this shuffle component support reliable storage - external to the lifecycle of the
    * executor host ? For example, writing shuffle data to a distributed filesystem or
    * persisting it in a remote shuffle service.
+   *
+   * Note: This method is for compatibility with older implementations,
+   * the newer implementation should not use this method.
```

Comment:
This comment addition is incorrect.
A shuffle implementation can either choose to be reliable irrespective of executor id, or choose to be reliable only for specific subset of executor id's - that is an implementation choice of the implementation, and not a function of newer vs older